programs/rodeos/wasm_ql_http.cpp

```diff
@@ -489,12 +492,37 @@ class http_session {
          queue_(*this) {}
 
    // Start the session
-   void run() { do_read(); }
+   void run() {
+      _timer.reset(new boost::asio::steady_timer(derived_session().stream.socket().get_executor()));
+      last_activity_timepoint = steady_clock::now();
+      start_socket_timer();
+      do_read(); 
+   }
 
  private:
    SessionType& derived_session() {
       return static_cast<SessionType&>(*this);
    }
+   
+   void start_socket_timer() 
+   {       
+      _timer->expires_after( http_config->idle_timeout_ms );
+      _timer->async_wait( [ this ]( beast::error_code ec ) {
+         if ( ec ){
+            return;
+         }
+         auto session_duration    = steady_clock::now() - last_activity_timepoint;
+         auto session_duration_ms = std::chrono::duration_cast<std::chrono::milliseconds>( session_duration ).count();
+         if ( session_duration_ms <= http_config->idle_timeout_ms.count() ){
```

Comment:
This can be simplified to:

```
         auto session_duration = steady_clock::now() - last_activity_timepoint;
         if ( session_duration <= http_config->idle_timeout_ms ){
```

Reply:
Thanks for catching this, fixed it.